### DISCUSSION TITLE
Would like to be able to set a shutdown command in Blueprints

### DISCUSSION THREAD

**Discussion by mixels** [gameserverapp/Platform#959] [Suggestions]
Many games have a console / RCON command that will shut the server down. I'd love to be able to specify a shutdown command for games that support it on the Blueprint which, if set, GSA would then invoke to shut the server process down, which in turn should cause the container to stop.

Some game servers need this because only the built in exit command causes the server to save its state before shutting down. For these games, killing the process by stopping the container causes loss of progress.

**Comment by birdman3131** [comment]
So i believe this is what the saveworld command is for. It should get sent before shutdown.

**Reply by mvaessen** [reply]
For folks reading this later: the __save command__ field is for inputting commands that can be used to save the world. Do not put anything else in there.

Only put a save command in the save command field.

**Reply by birdman3131** [reply]
Sorry about that. What was going through my head but failed to make it out was that afaik sending the save command is part of the shutdown sequence and thus an explicit shutdown command should not be needed.

**Comment by mvaessen** [comment]
Going to reject this one, because I do not agree with the reasoning. 

A `save` command will save the state before shutdown, this is the recommended method at this moment. 
Happy to review if we come across a situation where the current approach does not work, at all.

**Reply by mixels** [reply]
The game in question doesn't have a save command, though. It only saves when an exit command is sent to the server.

**Reply by mvaessen** [reply]
The game you're referring too (conan) has been on GSA for many years. There are no problem with save data getting lost, on shutdown. So based on that do I see no reason to change the approach.

As said, happy to review if there is indeed an issue with games that experience data-loss during shutdown. But so far I haven't come across any as of yet.
